diff -rc alpine-2.22/alpine/send.c alpine-2.22.count/alpine/send.c
*** alpine-2.22/alpine/send.c	2020-01-19 01:32:20.126476758 -0700
--- alpine-2.22.count/alpine/send.c	2020-01-19 01:34:58.323433969 -0700
***************
*** 5233,5243 ****
--- 5233,5245 ----
  	{'c', 'c', "C", N_("Confirm")},
  	{'n', 'n', "N", N_("No")},
  	{'y', 'y', "", ""},
+ 	{'t', 't', "T", N_("CounT")},
  	{-1, 0, NULL, NULL}
      };
  
      ps_global->redrawer = redraw_pico;
      fix_windsize(ps_global);
+     pbf->curpos[0] = '\0';
      
      while(1){
  	rv = radio_buttons(prompt, -FOOTER_ROWS(ps_global), opts,
***************
*** 5250,5261 ****
  	    q_status_message(SM_INFO, 1, 3, _(" Type \"C\" to cancel message "));
  	    display_message('x');
  	}
  	else
  	  break;
      }
  
      ps_global->redrawer = redraw;
!     return(rstr);
  }
  
  
--- 5252,5267 ----
  	    q_status_message(SM_INFO, 1, 3, _(" Type \"C\" to cancel message "));
  	    display_message('x');
  	}
+ 	else if(rv == 't'){
+ 	    showcpos(1,0);
+ 	    break;
+ 	}
  	else
  	  break;
      }
  
      ps_global->redrawer = redraw;
!     return(pbf->curpos[0] ? pbf->curpos : rstr);
  }
  
  
diff -rc alpine-2.22/pico/display.c alpine-2.22.count/pico/display.c
*** alpine-2.22/pico/display.c	2020-01-19 01:32:18.428510798 -0700
--- alpine-2.22.count/pico/display.c	2020-01-19 01:34:58.325433955 -0700
***************
*** 1806,1811 ****
--- 1806,1815 ----
      menu_yesno[6].label = N_("Cancel");
      menu_yesno[7].name  = "N";
      menu_yesno[7].label = (dflt == FALSE) ? "[" N_("No") "]" : N_("No");
+     if(Pmaster && Pmaster->onctrlc){
+       menu_yesno[8].name  = "T";
+       menu_yesno[8].label = N_("counT");
+     }
      wkeyhelp(menu_yesno);		/* paint generic menu */
      sgarbk = TRUE;			/* mark menu dirty */
      if(Pmaster && curwp)
***************
*** 1883,1888 ****
--- 1887,1900 ----
  		km_popped++;
  		break;
  	    }
+ 
+ 	  case 'T':
+ 	  case 't':
+ 	    if(Pmaster && Pmaster->onctrlc){
+ 	      pputs_utf8(_("counT"), 1);
+ 	      rv = COUNT;
+ 	      break;
+ 	    }
  	    /* else fall through */
  
  	  default:
diff -rc alpine-2.22/pico/pico.c alpine-2.22.count/pico/pico.c
*** alpine-2.22/pico/pico.c	2020-01-19 01:32:18.356512247 -0700
--- alpine-2.22.count/pico/pico.c	2020-01-19 01:34:58.325433955 -0700
***************
*** 674,685 ****
      result = "";
  
      Pmaster->arm_winch_cleanup++;
      if(Pmaster->canceltest){
          if(((Pmaster->pine_flags & MDHDRONLY) && !any_header_changes())
  	  || (result = (*Pmaster->canceltest)(redraw_pico_for_callback))){
- 	    pico_all_done = COMP_CANCEL;
  	    emlwrite(result, NULL);
  	    Pmaster->arm_winch_cleanup--;
  	    return(TRUE);
  	}
  	else{
--- 674,692 ----
      result = "";
  
      Pmaster->arm_winch_cleanup++;
+     Pmaster->onctrlc++;
      if(Pmaster->canceltest){
          if(((Pmaster->pine_flags & MDHDRONLY) && !any_header_changes())
  	  || (result = (*Pmaster->canceltest)(redraw_pico_for_callback))){
  	    emlwrite(result, NULL);
  	    Pmaster->arm_winch_cleanup--;
+ 	    if(Pmaster->curpos[0]){
+ 	       curwp->w_flag |= WFMODE;		/* and modeline so we  */
+ 	       sgarbk = TRUE;			/* redraw the keymenu  */
+ 	       pclear(term.t_nrow - 1, term.t_nrow + 1);
+ 	       return(FALSE);
+ 	    }
+ 	    pico_all_done = COMP_CANCEL;
  	    return(TRUE);
  	}
  	else{
***************
*** 708,713 ****
--- 715,726 ----
  	emlwwrite(_("Cancel Cancelled"), NULL);
  	break;
  
+       case COUNT:
+ 	showcpos(1,0);
+ 	emlwrite(Pmaster->curpos, NULL);
+ 	Pmaster->onctrlc--;
+ 	break;
+ 
        default:
  	mlerase();
      }
diff -rc alpine-2.22/pico/pico.h alpine-2.22.count/pico/pico.h
*** alpine-2.22/pico/pico.h	2020-01-19 01:32:18.309513193 -0700
--- alpine-2.22.count/pico/pico.h	2020-01-19 01:34:58.326433948 -0700
***************
*** 209,219 ****
--- 209,221 ----
      PCOLORS *colors;                    /* colors for titlebar and keymenu */
      void  *input_cs;			/* passed to mbtow() via kbseq() */
      long   pine_flags;			/* entry mode flags */
+     char  curpos[80];			/* where are we now? */
      /* The next few bits are features that don't fit in pine_flags      */
      /* If we had this to do over, it would probably be one giant bitmap */
      unsigned always_spell_check:1;      /* always spell-checking upon quit */
      unsigned strip_ws_before_send:1;    /* don't default strip bc of flowed */
      unsigned allow_flowed_text:1;    /* clean text when done to keep flowed */
+     unsigned onctrlc;			/* are we on ctrl-c command? */
      int   (*helper)();			/* Pine's help function  */
      int   (*showmsg)();			/* Pine's display_message */
      UCS   (*suspend)();			/* Pine's suspend */
diff -rc alpine-2.22/pico/random.c alpine-2.22.count/pico/random.c
*** alpine-2.22/pico/random.c	2020-01-19 01:32:18.280513777 -0700
--- alpine-2.22.count/pico/random.c	2020-01-19 01:34:58.326433948 -0700
***************
*** 74,80 ****
  	    thisline+1, lines+1, (int)((100L*(thisline+1))/(lines+1)),
  	    nbc, nch, (nch) ? (int)((100L*nbc)/nch) : 0);
  
!     emlwrite(buffer, NULL);
      return (TRUE);
  }
  
--- 74,83 ----
  	    thisline+1, lines+1, (int)((100L*(thisline+1))/(lines+1)),
  	    nbc, nch, (nch) ? (int)((100L*nbc)/nch) : 0);
  
!     if(Pmaster)
!       strcpy(Pmaster->curpos, buffer);
!     else
!       emlwrite(buffer, NULL);
      return (TRUE);
  }
  
